### PR TITLE
V3 flatbuffers

### PR DESCRIPTION
**Pull request by pipu**
1.  Add flat buffers
2.  Add files::NodeReader, SingleNodeReader, SpriteReader, ParticleReader, GameMapReader, ProjectNodeReader, NodeReaderProtol, NodeReaderDefine, FlatBufferSerialize, CSParseBinary_generated.h

### CODE REVIEW

**Comment by joewan** [comment]
great!

**Comment by minggo** [comment]
[ci rebuild]

**Comment by minggo** [comment]
[ci rebuild]

File: cocos/Android.mk [hunk 1]
```diff
@@ -273,3 +273,4 @@ $(call import-module,Box2D)
 $(call import-module,curl/prebuilt/android)
 $(call import-module,websockets/prebuilt/android)
 $(call import-module,protobuf-lite)
+$(call import-module,flatbuffers)
```
Comment (minggo) [inline cocos/Android.mk:276]: Where is the module `flatbuffers` defined?

**Comment by ricardoquesada** [comment]
First of all, thanks for creating this PR. It is indeed needed. Thanks!

But, please, you should know that the PR broke the build and our users (and Microsoft and I) can't work if the build is broken.

I reverted this PR. Please, update the PR with the needed fixes in order to make it compile. Thanks!

**Comment by pjc0247** [comment]
@ricardoquesada 
1. download [flatbuffers](https://github.com/google/flatbuffers), and place it to external/flatbuffers
2. change this [line](https://github.com/cocos2d/cocos2d-x/blob/fa22b126285674e3e0d0f2d5f2f39c56987b483e/cocos/editor-support/cocostudio/WidgetReader/NodeReader/NodeReader.cpp#L81) to 

``` C++
Color4B color = (255, 255, 255, 255);
```

for non vs2013 users
(vs2012 dosn't support initializer-lists)

**Comment by minggo** [comment]
I merged it because only linux is broken.
And we know linux is broken by another PR.
And this pull request changed a lot, it is easy to cause conflict.

**Comment by minggo** [comment]
@pipu 
Please send it again.
Thanks.

**Comment by ricardoquesada** [comment]
Actually this PR broke everything: Mac, Windows and Linux :(

I tested it this morning and I couldn't compile anything. Perhaps the dependencies weren't updated or something else, but branch `v3` was totally broken.

**Comment by minggo** [comment]
So there is problem in jenkins.
It shows all platforms other than linux build ok.

**Comment by ricardoquesada** [comment]
Yes, last week Jenkins was broken,  I merged some PR that broke the build, but I didn't notice that because the merge button was green.

Probably the same happened to you.

I told Shun that the HIGHEST priority of all, is to make sure the Jenkins always work to prevent these things.

**Pull request merged**